File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/datatype/ShortType.kt
package dynabuffers.ast.datatype

import dynabuffers.api.IRegistry
import dynabuffers.api.ISerializable
import dynabuffers.api.IType
import java.nio.ByteBuffer

class ShortType : IType, ISerializable {

    override fun size(value: Any, registry: IRegistry) = 2

    override fun serialize(value: Any, buffer: ByteBuffer, registry: IRegistry) {
        buffer.putShort(short(value))
    }

    override fun deserialize(buffer: ByteBuffer, registry: IRegistry) = buffer.short

    override fun supports(value: Any): Boolean {
        return value is Number && value.toDouble() <= Short.MAX_VALUE && value.toFloat() >= Short.MIN_VALUE
    }

    private fun short(obj: Any) = obj.toString().toShort()

}

File: dynabuffers-python/dynabuffers/ast/datatype/ShortType.py
from dynabuffers.api.ISerializable import ISerializable


class ShortType(ISerializable):

    def size(self, value, registry):
        return 2

    def serialize(self, value, buffer, registry):
        buffer.putShort(value if isinstance(value, int) else int(value))

    def deserialize(self, buffer, registry):
        return buffer.getShort()
